### ISSUE TITLE
Pass existing injector into boostrap function?

### ISSUE THREAD

**Issue by bcherny** [angular/angular.js#15784]
I have a scenario where I declare a module, bootstrap it, then declare a 2nd module, and bootstrap the 2nd module. My constraint is that the 1st module's injector must be created before the 2nd module is declared.

However, because I bootstrap twice, I end up with 2 injectors. What do you guys think of allowing an optional `injector` property on the `config` object passed into `bootstrap()`, which can be used to allow reusing an existing injector instead of creating a new one when bootstrapping an app.

If that sounds reasonable, I would be happy to put up a PR.

Thanks!

**Comment by Narretz** [comment]
I personally think that adds a new option without a strong / frequent use case. Why do you need this setup with the different modules?

**Comment by bcherny** [comment]
Agreed - it's an exotic use case, but without it I can't implement what I need.

It's for https://github.com/bcherny/ngimport. The idea is to make all built in services ($http, $log, etc.) importable (eg. `import {$http} from 'ngimport'`). The code works by retrieving these services from the $injector, then re-exporting them via regular `export`s.

In order for services imported from ngimport to be defined, the injector must be created already, which means the app must be bootstrapped already.

There are 3 ways out:

1. Services imported from ngimport are undefined until you bootstrap your app
2. Ngimport self-bootstraps, and exports an injector that you can pass into your own bootstrap function

Is there some other way to implement this?

**Comment by gkalpak** [comment]
You might be able to hack around it by monkey-patching `$provide` and/or `$injector` to return the instances created in another injector. In any case, note that instantiating providers or services may have side-effects. So the app may behave differently with/without `ngimport` if you eagerly instantiate some providers/services that might not be necessary.
